aptos-move/framework/aptos-framework/doc/transaction_context.md

```diff
@@ -5,14 +5,143 @@
 
 
 
+-  [Struct `UUID`](#0x1_transaction_context_UUID)
+-  [Constants](#@Constants_0)
+-  [Function `get_txn_hash`](#0x1_transaction_context_get_txn_hash)
+-  [Function `create_unique_address`](#0x1_transaction_context_create_unique_address)
+-  [Function `create_unique_addr`](#0x1_transaction_context_create_unique_addr)
 -  [Function `get_script_hash`](#0x1_transaction_context_get_script_hash)
--  [Specification](#@Specification_0)
-    -  [Function `get_script_hash`](#@Specification_0_get_script_hash)
+-  [Function `create_uuid`](#0x1_transaction_context_create_uuid)
+-  [Specification](#@Specification_1)
+    -  [Function `get_script_hash`](#@Specification_1_get_script_hash)
 
 
-<pre><code></code></pre>
+<pre><code><b>use</b> <a href="../../aptos-stdlib/../move-stdlib/doc/features.md#0x1_features">0x1::features</a>;
+</code></pre>
+
+
+
+<a name="0x1_transaction_context_UUID"></a>
+
+## Struct `UUID`
+
+A wrapper denoting universally unique identifer (UUID)
+for storing an address
+
+
+<pre><code><b>struct</b> <a href="transaction_context.md#0x1_transaction_context_UUID">UUID</a> <b>has</b> drop, store
+</code></pre>
+
+
+
+<details>
+<summary>Fields</summary>
+
+
+<dl>
+<dt>
+<code>unique_address: <b>address</b></code>
+</dt>
+<dd>
+
+</dd>
+</dl>
+
+
+</details>
+
+<a name="@Constants_0"></a>
+
+## Constants
+
+
+<a name="0x1_transaction_context_EUUID_NOT_SUPPORTED"></a>
+
+UUID feature is not supported.
+
+
+<pre><code><b>const</b> <a href="transaction_context.md#0x1_transaction_context_EUUID_NOT_SUPPORTED">EUUID_NOT_SUPPORTED</a>: u64 = 3;
+</code></pre>
+
+
+
+<a name="0x1_transaction_context_get_txn_hash"></a>
+
+## Function `get_txn_hash`
+
+Return the transaction hash of the current transaction
 
 
+<pre><code><b>public</b> <b>fun</b> <a href="transaction_context.md#0x1_transaction_context_get_txn_hash">get_txn_hash</a>(): <a href="../../aptos-stdlib/../move-stdlib/doc/vector.md#0x1_vector">vector</a>&lt;u8&gt;
+</code></pre>
+
+
+
+<details>
+<summary>Implementation</summary>
+
+
+<pre><code><b>public</b> <b>native</b> <b>fun</b> <a href="transaction_context.md#0x1_transaction_context_get_txn_hash">get_txn_hash</a>(): <a href="../../aptos-stdlib/../move-stdlib/doc/vector.md#0x1_vector">vector</a>&lt;u8&gt;;
+</code></pre>
+
+
+
+</details>
+
+<a name="0x1_transaction_context_create_unique_address"></a>
+
+## Function `create_unique_address`
+
+Return a universally unique identifier (of type address) generated
+by hashing the transaction hash of this transaction and a sequence number
```

Comment:
do we need some sort of domain separation?

Reply:
We have domain separation. I will change this comment to reflect that.